### PR TITLE
Update serverless-init Cloud Run service to support cloud run functions via Cloud Functions UI

### PR DESCRIPTION
**Pull request by nina9753**
### What does this PR do?

Update serverless-init Cloud Run service to support Cloud Run functions
This adds the other environmental variables returned by cloud run and cloud run functions into the tag array. `K_CONFIGURATION`, `FUNCTION_SIGNATURE_TYPE`, `FUNCTION_TARGET`
This also updates the metric prefix to be `gcp.cloudfunction` and the origin is `cloudfunction`

Customers will now need to add the `FUNCTION_TARGET` environment variable to their cloud run functions container

### Motivation

Google Cloud Run Function [now supports sidecars](https://cloud.google.com/blog/products/serverless/google-cloud-functions-is-now-cloud-run-functions) under cloud run 

### Additional Notes

Currently blocks PR #29057 which will update the origin to `cloudrunfunction` when we make this product GA 

### Possible Drawbacks / Trade-offs

Currently, this approach will work for all runtimes except Go. Which does not have any [out of the box environmental variables ](https://cloud.google.com/functions/docs/running/function-frameworks#functions-local-ff-configure-go)we could use to mark a service as a cloud function source deploy. We will need to force the customer to add  `FUNCTION_TARGET` to the variables during setup so that in datadog everything is tagged correctly 

### Describe how to test/QA your changes

added a new test case `TestGetCloudRunFunctionTagsWithEnvironmentVariables` you can also run all test locally by running `go test -tags "test" -v ./cmd/serverless-init/...`

currently deployed as [gcp-run-function-rc3](https://github.com/datadog/datadog-lambda-extension/pkgs/container/datadog-lambda-extension%2Fserverless-init/275195238?tag=gcp-run-function-rc3) and deployed to our team project `datadog-serverless-gcp-demo` inside the [artifact registry tag serverless-init-sidercar here](https://console.cloud.google.com/artifacts/docker/datadog-serverless-gcp-demo/us/gcr.io/serverless-init-sidercar/sha256:ee6c43e43e85d518b19bff209bc489d4f05bd5fe24b40f66fe475a3609817e8f?project=datadog-serverless-gcp-demo)
You can view the test service in the serverless org [here](https://ddserverless.datadoghq.com/functions?search=service_name%3Aself-monitoring-sidecar-python-run-function&cloud=gcp&entity_view=cloud_run&fromUser=false&graphType=span_list&panel_end=1726671635946&panel_paused=false&panel_start=1726066835946&panel_tab=traces&selection=cloud_run%2Bself-monitoring-sidecar-python-run-function%2Bus-central1%2Bdatadog-serverless-gcp-demo&shouldShowLegend=true&sort=time&sort_cloud_run=service_request_rate&start=1726066593535&end=1726671393535&paused=false)

### CODE REVIEW

File: cmd/serverless-init/cloudservice/cloudrun.go [hunk 1]
```diff
@@ -64,3 +99,9 @@ func isCloudRunService() bool {
 	_, exists := os.LookupEnv(ServiceNameEnvVar)
 	return exists
 }
+
+func isCloudRunFunction() bool {
+	_, cloudRunFunctionMode := os.LookupEnv(functionTargetEnvVar)
+	_ = log.Warn(fmt.Sprintf("cloud function mode SET TO: %t", cloudRunFunctionMode))
```
Comment (purple4reina) [inline cmd/serverless-init/cloudservice/cloudrun.go:105]: Does this really need to be assigned to `_` in order for it to work? I'm guessing no.

Also, I'm thinking this is more a debug level log than a warn. What are your thoughts on that?

**Review by purple4reina — COMMENTED** [review]
Had just one question for you, otherwise looks good!

File: cmd/serverless-init/cloudservice/cloudrun.go (hunk 1 above)
Comment (nina9753) [reply cmd/serverless-init/cloudservice/cloudrun.go]: Great call out! I updated the log level to debug. Line 104 returns 2 variables to i need to keep the leading `_` as I am ignoring this return value

**Review by purple4reina — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

**Pull request merged**